vignettes/validation_figure_tests.Rmd

```diff
@@ -0,0 +1,89 @@
+---
+title: "Validation: Figure Tests"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{validation_figure_tests}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(valtools)
+library(testthat)
+```
+
+Effective test cases cover a broad spectrum of use cases of the code intended to be validated. 
+More often than not, this is confirming that an input returns a specific or set of specific values.
+
+However, not all functions are used for their value-generation.
+In fact, there are a number of functions that are used for their side effects, such as plot generation. 
```

Comment:
{testthat} identifies the threshold as the pain point of programmatically recording expected behavior with code. I think we should adopt the {testthat} perspective instead of re-framing our own definition, because the "new" factor here is using snapshots in {valtools}